File: Practica2/c++/mul.cc
/*****************************************************************
 * File:   mul.cc
 * Author: Claudia Pavón (852698) y Lucia Morales (816906)
 * Date:   23 febrero 2023
 * Coms:   Implementacion de la clase Mul, subclase de Instruccion
*****************************************************************/
#include "mul.h"

Mul::Mul() : Instruccion(){
};

// Desapila dos valores de la pila y apila su producto
int Mul::ejecutar(stack<int>& pila, const int contador){
   int a = pila.top();
   pila.pop();
   int b = pila.top();
   pila.pop();
   pila.push(a*b);

   return contador + 1;
}

// Devuelve una cadena de texto que representa la instrucción
string Mul::nombre() const {
   return "mul";
}

File: Practica2/java/Mul.java
/*****************************************************************
 * File:   Mul.java
 * Author: Claudia Pavón (852698) y Lucia Morales (816906)
 * Date:   23 febrero 2023
 * Coms:   Implementación de la clase Mul, subclase de Instruccion
*****************************************************************/
import java.util.*;

public class Mul extends Instruccion {
   protected int c;
   public Mul(){}

   @Override
   public int ejecutar(Stack<Integer> pila, int contador){
      int a = pila.peek();
      pila.pop();
      int b = pila.peek();
      pila.pop();
      pila.push(a*b);

      return contador + 1;
   }
   @Override
   public String nombre() {
      return "mul";
   }
}